fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/SocialMediaRepository.kt

```diff
@@ -0,0 +1,353 @@
+                .subscribe({
+                  Timber.tag(TAG).d("Successfully persisted deleted reaction %s", deleted.id)
+                }) { err: Throwable ->
+                  Timber.tag(TAG).e(err, "Error in persisting deleted reaction %s", deleted.id)
+                })
+      }
+      return true
+    }
+
+    fun getChirpsSubject(): Observable<Set<MessageID>> {
+      return chirpsSubject
+    }
+
+    @Throws(UnknownChirpException::class)
+    fun getChirp(id: MessageID): Observable<Chirp> {
+      return chirpSubjects[id] ?: throw UnknownChirpException(id)
+    }
+
+    @Throws(UnknownChirpException::class)
+    fun getReactions(chirpId: MessageID): Observable<Set<Reaction>> {
+      return reactionSubjectsByChirpId[chirpId] ?: throw UnknownChirpException(chirpId)
+    }
+
+    /**
+     * Load in memory the chirps and their respective reactions from the disk only when the user
+     * wants to inflate the chirps adapter. It can be done only once per LAO, as during the
+     * execution everything is also stored in memory.
+     */
+    private fun loadStorage() {
+      repository.disposables.add(
+          repository.chirpDao
+              .getChirpsByLaoId(laoId)
+              .subscribeOn(Schedulers.io())
+              .observeOn(AndroidSchedulers.mainThread())
+              .subscribe({ chirpsList: List<Chirp> ->
+                chirpsList.forEach(
+                    Consumer outer@{ chirp: Chirp ->
+                      // Do not retrieve deleted chirps
+                      if (chirp.isDeleted) {
+                        return@outer
+                      }
+                      // Load the chirp into the memory
+                      add(chirp)
+                      Timber.tag(TAG).d("Retrieved from db chirp %s", chirp.id)
+                      // When retrieving the chirp also retrieve its reactions
+                      repository.disposables.add(
+                          repository.reactionDao
+                              .getReactionsByChirpId(chirp.id)
+                              .subscribeOn(Schedulers.io())
+                              .observeOn(AndroidSchedulers.mainThread())
+                              .subscribe({ reactionsList: List<Reaction> ->
+                                reactionsList.forEach(
+                                    Consumer inner@{ reaction: Reaction ->
+                                      // Do not retrieve deleted reactions
+                                      if (reaction.isDeleted) {
+                                        return@inner
+                                      }
+                                      // Load the reaction into the memory
+                                      addReaction(reaction)
+                                      Timber.tag(TAG)
+                                          .d("Retrieved from db reaction %s", reaction.id)
+                                    })
+                              }) { err: Throwable ->
```

Comment:
same question, are you sure this is correct and you don't mean to do the following?  🤔 
```suggestion
                              }, { err: Throwable ->
```